api/search.rb

```diff
@@ -1,104 +1,138 @@
-          query do
-            match [:title, :body], search_text, :operator => "AND"
-            filtered do
-              filter :term, :commentable_id => local_params["commentable_id"] if local_params["commentable_id"]
-              filter :terms, :commentable_id => local_params["commentable_ids"].split(",") if local_params["commentable_ids"]
-              filter :term, :course_id => local_params["course_id"] if local_params["course_id"]
-              filter :or, [
-                {:not => {:exists => {:field => :context}}},
-                {:term => {:context => context}}
-              ]
+  filters.push({or: [
+      {not: {exists: {field: :context}}},
+      {term: {context: context}}
+  ]})
+
+  unless group_ids.empty?
+    filters.push(
+        {
+            or: [
+                {:not => {:exists => {:field => :group_id}}},
+                {:terms => {:group_id => group_ids}}
+            ]
+        }
+    )
+  end
 
-              if not group_ids.empty?
-                if group_ids.length > 1
-                  group_id_criteria = {:terms => {:group_id => group_ids}}
-                else
-                  group_id_criteria = {:term => {:group_id => group_ids[0]}}
-                end
+  self.class.trace_execution_scoped(['Custom/get_search_threads/es_search']) do
+    body = {
+        size: CommentService.config['max_deep_search_comment_count'].to_i,
+        sort: [
+            {updated_at: :desc}
+        ],
+        query: {
+            multi_match: {
+                query: search_text,
+                fields: [:title, :body],
+                operator: :AND
+            },
+            filtered: {
+                filter: {
+                    and: filters
+                }
+            }
+        }
+    }
 
-                filter :or, [
-                  {:not => {:exists => {:field => :group_id}}},
-                  group_id_criteria
-                ]
-              end
+    response = Elasticsearch::Model.client.search(index: Content::ES_INDEX_NAME, body: body)
 
-            end
-          end
-          sort do
-            by "updated_at", "desc"
-          end
-          size CommentService.config["max_deep_search_comment_count"].to_i
-        end
-        thread_ids = Set.new
-        search.results.each do |content|
-          case content.type
-          when "comment_thread"
-            thread_ids.add(content.id)
-          when "comment"
-            thread_ids.add(content.comment_thread_id)
-          end
-        end
-        thread_ids
+    thread_ids = Set.new
+    response['hits']['hits'].each do |hit|
+      case hit['_type']
+        when CommentThread.document_type
+          thread_ids.add(hit['_id'])
+        when Comment.document_type
+          thread_ids.add(hit['_source']['comment_thread_id'])
+        else
+          # There shouldn't be any other document types. Nevertheless, ignore them, if they are present.
+          next
```

Comment:
@clintonb: Please ignore.  I'm going to leave this as-is.  I don't see the message helping much.